### PR TITLE
Migrate existing tests for Codaveri v2 integration

### PR DESCRIPTION
**Pull request by adi-herwana-nus**
## Background
This PR updates test cases to align with new Codaveri v2 integration changes introduced by [this PR](https://github.com/Coursemology/coursemology2/pull/7379)

## Changes
- update jstest
- update all rspec codaveri endpoint mocks/fixtures to align with current Codaveri v2 schema
- updated spec from mocking connect_to_codaveri function to using [Excon.stub](https://github.com/excon/excon?tab=readme-ov-file#stubs)
- centralized codaveri api mocks into separate directory
- deleted outdated mocks

## TODO
- Write tests for new polling endpoint behavior in backend job
- Finish removing dead v1 integration code (`codaveri_api_service.rb`, deprecated in favor of `codaveri_async_api_service.rb`)

### CODE REVIEW

File: spec/rails_helper.rb [hunk 1]
```diff
@@ -9,6 +9,8 @@
 ENV['KEYCLOAK_ISS'] ||= 'http://localhost:8443/realms/coursemology_test'
 ENV['KEYCLOAK_AUD'] ||= 'account'
 ENV['KEYCLOAK_REALM'] ||= 'coursemology_test'
+# All the codaveri endpoints are mocked, so this URL should never be called
```
Comment (cysjonathan) [inline spec/rails_helper.rb:12]: If it's not called, why do we still need to specify it here?

Comment (cysjonathan) [reply spec/rails_helper.rb:12]: Apart from this, rest LGTM, update to a dummy value then we can merge it in.

Comment (adi-herwana-nus) [reply spec/rails_helper.rb:12]: Done.

Comment (cysjonathan) [reply spec/rails_helper.rb:12]: can you squash your last 2 commits?
Also the env is still pointing to the live url instead of a dummy one.

Comment (adi-herwana-nus) [reply spec/rails_helper.rb:12]: Done.

Had to make one more change: coursemology docker container test requires the real docker apis (or ping real docker service locally), so I had it set Excon to not mock requests

**Review by cysjonathan — APPROVED** [review]

**Pull request merged**